crates/pipeline_manager/src/db/mod.rs

```diff
@@ -817,6 +819,86 @@ impl Storage for ProjectDB {
     async fn delete_service(&self, tenant_id: TenantId, service_name: &str) -> Result<(), DBError> {
         Ok(service::delete_service(self, tenant_id, service_name).await?)
     }
+
+    async fn new_service_probe(
+        &self,
+        tenant_id: TenantId,
+        service_id: ServiceId,
+        id: Uuid,
+        request: ServiceProbeRequest,
+        created_at: &DateTime<Utc>,
+        txn: Option<&Transaction<'_>>,
+    ) -> Result<ServiceProbeId, DBError> {
+        Ok(
+            service::new_service_probe(self, tenant_id, service_id, id, request, created_at, txn)
+                .await?,
+        )
+    }
+
+    async fn update_service_probe_set_running(
```

Comment:
Better to have just update_service_probe() and pass the state like `Status::Running` as an argument. It allows you to evolve the API better over time, especially when new statuses potentially emerge (which it likely will looking at the current API).

Reply:
Hmm, I thought it would be simpler to keep it two separate functions to avoid complex logic such as exists in [update_program](https://github.com/feldera/feldera/blob/60169ef129380669b34cf1d2d3664659247cfe87/crates/pipeline_manager/src/db/program.rs#L222C21-L222C35). One other minor reason I had was that there is no wrong way to call the functions with them separated like this (e.g., updating finished when setting status to running). I reasoned an internal switch statement would be as the same as two functions. One last minor reason is that there is generally tricky part where None is used sometimes to indicate "No value (NULL)" and "Don't update". In this case, the timestamp will never be updated to NULL afterwards, but it requires additional comments to convey this is the case (the meaning of None). What do you think?

Reply:
It's fine for now, but so far in the code, it's been the case that adding new states bloats the interface with new methods that only differ by the value of one argument. I was hoping to preempt such a change. :) I'll leave it to your call.

The solution to the `None` issue is to actually use `Option<Option<T>>`, the outer one stating whether to update or not and the inner one having the actual value.

Reply:
Yeah I'm not a fan either, but if the API needs to optionally update a field that takes an Option<T> I don't see another way.

Reply:
As the database API is internal, and I don't anticipate that a probe will gain more statuses in the foreseeable future beyond current pending, running, success and failure, I'll leave it two separate functions for now. If a third function is needed in future, we can re-evaluate and possible simplify the code by folding it into one.